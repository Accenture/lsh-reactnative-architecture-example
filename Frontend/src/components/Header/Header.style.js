const headerStyles = {
  viewStyle: {
    backgroundColor: '#F8F8F8',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: 60,
    paddingTop: 15,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    elevation: 2,
    position: 'relative',
    flexDirection: 'row',
  },
  textStyle: {
    fontSize: 20,
  },
  hamburgerStyle: {
    flexDirection: 'row',
    padding: 10,
  },
};
export default headerStyles;
