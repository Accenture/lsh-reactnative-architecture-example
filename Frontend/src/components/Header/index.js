import React from 'react';
import PropTypes from 'prop-types';
import { Text, View, TouchableOpacity } from 'react-native';
import { Icon } from 'react-native-elements';
import translate from 'Frontend/src/commons/Language';
import styles from './Header.style';

const Header = ({ onPress, headerText, hideDrawer }) => {
  const { textStyle, viewStyle, hamburgerStyle } = styles;
  return (
    <View style={viewStyle}>
      <View style={hamburgerStyle}>
        <TouchableOpacity onPress={onPress}>
          {!hideDrawer && <Icon name="menu" />}
        </TouchableOpacity>
      </View>
      <Text style={textStyle}>{translate(headerText)}</Text>
      <TouchableOpacity onPress={onPress}>
        {!hideDrawer && <Icon name="notifications-none" />}
      </TouchableOpacity>
    </View>
  );
};

Header.defaultProps = {
  headerText: 'Header',
  onPress: PropTypes.func.isRequired,
  hideDrawer: false,
};

Header.propTypes = {
  headerText: PropTypes.string,
  onPress: PropTypes.func,
  hideDrawer: PropTypes.bool,
};

export default Header;
