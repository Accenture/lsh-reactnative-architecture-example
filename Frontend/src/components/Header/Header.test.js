import React from 'react';
import { shallow } from 'enzyme';
import Header from './index';

describe('Tests header component', () => {
  it('matches snapshot', () => {
    const comp = shallow(<Header />);
    expect(comp).toMatchSnapshot();
  });
});
