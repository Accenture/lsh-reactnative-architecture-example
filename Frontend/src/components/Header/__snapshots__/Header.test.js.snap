// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests header component matches snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "elevation": 2,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-between",
      "paddingTop": 15,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
    }
  }
>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Icon
        color="black"
        name="menu"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={24}
        underlayColor="white"
      />
    </TouchableOpacity>
  </Component>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Header
  </Text>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Icon
      color="black"
      name="notifications-none"
      raised={false}
      reverse={false}
      reverseColor="white"
      size={24}
      underlayColor="white"
    />
  </TouchableOpacity>
</Component>
`;
